### ISSUE TITLE
No albums detected

### ISSUE THREAD

**Issue by bigZos** [Salvoxia/immich-folder-album-creator#79]
Immich is successfully scanning my external libraries and I see all the folders and assets in the Immich UI. However, when I run the script, it reports:
 
Detected Immich server version 1.122.0
Requesting all assets
72453 photos found
Sorting assets to corresponding albums using folder name
**0 albums identified**
Album list: []

I have confirmed that:

- I’m using the correct root_path inside the container 
- Assets appear correctly in Immich and are accessible via the external library paths.
- I’ve tried different root_path values and removed filtering logic to ensure I’m not excluding folders unintentionally.

Despite this, the script still doesn’t detect any albums. Can you help me figure out what I might be doing wrong or what additional steps I need to take to align the script’s expectations with Immich’s external library paths?

**Comment by Salvoxia** [comment]
I'm sure we can figure out what's going on.
Could you please share the structure of your external library, which path you mounted it to the Immich container under and the exact call you used to invoke the script?

**Comment by bigZos** [comment]
Sure, here are the details:

External Library Structure (on the host machine):
I have a Windows drive W:\01. Albums that contains multiple subfolders like BOSTON, Family, etc. For example:

W:\01. Albums
├── BOSTON
│   ├── event1
│   │   └── photo.jpg
│   └── event2
│       └── photo2.jpg
├── Family
│   ├── 2022
│   │   └── Summer
│   │       └── Beach
│   │           └── photo3.jpg
└── ... other folders ...

Docker Compose Configuration:
In my docker-compose.yml, I’ve mounted these folders into the container under /import like so:

immich-server:
  volumes:
    - "W:/01. Albums/BOSTON:/import/BOSTON:ro"
    - "W:/01. Albums/Family:/import/Family:ro"
    # ... and so on for each folder ...

Inside the Immich container, these appear as /import/BOSTON, /import/Family, etc. In the Immich UI, I added external libraries pointing directly to each path (e.g. /import/BOSTON).

Script Invocation: I’ve tried several variations. For example:

I tried specifying a single folder
python immich_auto_album.py --root_path "/import/BOSTON" 

I’ve also tried using python immich_auto_album.py --root_path "/import" to cover all folders under /import.

I’ve attempted to run it directly from VS Code by pressing “Run” on immich_auto_album.py with the same parameters defined.

In all cases, the script finishes without errors but outputs:

0 albums identified
Album list: []

Immich itself displays all assets and folders correctly in its folder view, so I know the external libraries are scanned and working inside Immich. I just can’t get the script to detect and create the albums.

Is there something off about my setup, or the root_path I’m using?

**Comment by Salvoxia** [comment]
Just real quick, I only looked at your script call for now, please try calling the script by specifying the root path as a positional argument:
`python immich_auto_album.py "/import" http://immichserver.exmaple.com/api YOURAPIKEY`

**Comment by bigZos** [comment]
Hey, thanks for getting back to me and offering your suggestions! I managed to get it working. It turns out I needed to properly handle multiple root paths as a list rather than as a single comma-separated string. Once I adjusted the argument parsing (using action='append' or splitting the root paths into a proper list), the script was able to detect my albums correctly.

i did it like this:
parser.add_argument("--root_path", 
                    help="One or more external library root paths in Immich", 
                    action='append', 
                    default=["/import/BOSTON",
                             "/import/Family",
                               etc...
                    ])

The script is awesome! it created all the albums in seconds and it worked beautifully i can see them all now in the UI

I’m all set now :) Really appreciate your time and help!

**Comment by Salvoxia** [comment]
Just for my understanding: You changed the default value for `--root-path` in the script itself? That change will be gone after an update.
However, this should work for you:
`python immich_auto_album.py --root-path "/import/BOSTON"  "/import/Family" "http://immichserver.exmaple.com/api" YOURAPIKEY`

Originally, the script only supported a single `root_path`, and that was passed as the first positional argument, followed by the Immich Server API URL and API Key. Later, support for multiple root paths was added. But to keep compatibility with existing installations, I decided not to remove the root path from the positional arguments, but to add an optional `--root-path` argument that can be repeated multiple times to add a second, third, etc.. root path.

But misunderstandings with regards to this have occurred multiple times lately, so maybe I should look at improving the documentation :wink:
